src/python_testing/TC_AVSM_2_12.py

```diff
@@ -78,26 +80,64 @@ def steps_TC_AVSM_2_12(self) -> list[TestStep]:
+                     "Verify that the DUT response contains a bool value. Store value as aMicrophoneAGCEnabled"),
+            TestStep(33, "TH writes MicrophoneAGCEnabled attribute with value set to !aMicrophoneAGCEnabled.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(34, "TH reads MicrophoneAGCEnabled attribute.",
+                     "Verify that the value is the same as was written in Step 33."),
+            TestStep(35, "TH reads ImageRotation attribute.",
+                     "Verify that the DUT response is not greater than 359. Store value as aImageRotation"),
+            TestStep(36, "TH writes ImageRotation attribute with a new value of aImageRotation.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(37, "TH reads ImageRotation attribute.", "Verify that the value is the same as was written in Step 36."),
+            TestStep(38, "TH reads ImageFlipHorizontal attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aImageFlipHorizontal"),
+            TestStep(39, "TH writes ImageFlipHorizontal attribute with value set to !aImageFlipHorizontal.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(40, "TH reads ImageFlipHorizontal attribute.",
+                     "Verify that the value is the same as was written in Step 39."),
+            TestStep(41, "TH reads ImageFlipVertical attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aImageFlipVertical"),
+            TestStep(42, "TH writes ImageFlipVertical attribute with value set to !aImageFlipVertical.",
                      "DUT responds with a SUCCESS status code."),
-            TestStep(22, "TH reads MicrophoneMuted attribute.", "Verify that the value is the same as was written in Step 21."),
-            TestStep(23, "TH reads LocalVideoRecordingEnabled attribute.",
+            TestStep(43, "TH reads ImageFlipVertical attribute.",
+                     "Verify that the value is the same as was written in Step 42."),
+            TestStep(44, "TH reads LocalVideoRecordingEnabled attribute.",
                      "Verify that the DUT response contains a bool value. Store value as aLocalVideoRecordingEnabled"),
-            TestStep(24, "TH writes LocalVideoRecordingEnabled attribute with value set to !aLocalVideoRecordingEnabled.",
+            TestStep(45, "TH writes LocalVideoRecordingEnabled attribute with value set to !aLocalVideoRecordingEnabled.",
                      "DUT responds with a SUCCESS status code."),
-            TestStep(25, "TH reads LocalVideoRecordingEnabled attribute.",
-                     "Verify that the value is the same as was written in Step 24."),
+            TestStep(46, "TH reads LocalVideoRecordingEnabled attribute.",
+                     "Verify that the value is the same as was written in Step 45."),
```

Comment:
We seem to be missing the last 3 of the writable attributes, viz., LocalSnapshotRecordingEnabled, StatusLightEnabled, and StatusLightBrightness.

Reply:
Coverage added